lesson9/lesson9.js

```diff
@@ -0,0 +1,44 @@
+const ul = document.getElementById('js-list');
+const listItems = [
+  { to: "bookmark.html", img: "bookmark-icon.png", alt: "画像1", text: "ブックマーク" },
+  { to: "message.html", img: "mail-icon.png", alt: "画像2", text: "メッセージ" }
+];
+
+function showLoadingImg() {
+  const loadingImg = document.createElement('img');
+  loadingImg.src = "loading-circle.gif";
+  loadingImg.id = 'loading-img'
+  ul.appendChild(loadingImg);
+}
+
+function deleteLoadingImg() {
+  const loading = document.getElementById('loading-img');
+  loading.remove();
+}
+
+function createList(item) {
+  const fragment = document.createDocumentFragment();
+  for (let i = 0; i <= item.length - 1; i++) {
+    const li = document.createElement('li');
+    const a = document.createElement('a');
+    const img = document.createElement('img');
+    a.textContent = item[i].text;
+    a.href = item[i].to;
+    img.src = item[i].img;
+    img.alt = item[i].alt;
+    fragment.appendChild(li).appendChild(a).appendChild(img);
+  }
+  ul.appendChild(fragment)
+}
+
+async function fetchListData() {
+  showLoadingImg()
+  const promise = new Promise((resolve) => {
+    setTimeout(() => resolve(listItems), 3000)
+  })
+  const item = await promise;
+  createList(item)
+  deleteLoadingImg()
+}
```

Comment:
[IMO]
I also think that it is not desirable for showLoadingImg() and deleteLoadingImg() to exist in fetchListData().
How about separating the parts that fetch the data?

Reply:
Thank you for your comment! As you said, I thought it was inappropriate to display / delete read images when acquiring data. Removed showLoadingImg () deleteLoadingImg () in fetchListData ().
https://github.com/rikokko/to_be_frontend/commit/89d057a129ace8e76ff92fe4f9f992fd68f8bb89

Reply:
I thought that it was not ideal for createList() to be included in fetchListData(),since createList() works to build the DOM rather than to fetch data.
I also thought it would be more readable to put createList() outside of fetchListData() and await fetchListData().

Reply:
Thank you for your comment.
Removed createList () from fetchListData ().

I couldn't solve the problem "I thought it would be easier to read if I waited for fetchListData ()", so I would appreciate your advice. I could only think of waiting 3 seconds for a list item with a timeout.

Reply:
For example, I was thinking of something like the following.

`function createList(item) {
  const fragment = document.createDocumentFragment();
  for (let i = 0; i <= item.length - 1; i++) {
    const li = document.createElement("li");
    const a = document.createElement("a");
    const img = document.createElement("img");
    a.textContent = item[i].text;
    a.href = item[i].to;
    img.src = item[i].img;
    img.alt = item[i].alt;
    fragment.appendChild(li).appendChild(a).appendChild(img);
  }
  ul.appendChild(fragment);
}

async function fetchListData() {
  return new Promise((resolve) => {
    setTimeout(() => resolve(listItems), 3000);
  });
}

async function asyncProcessing() {
  showLoading();
  const value = await fetchListData();
  removeLoading();
  createList(value);
}

asyncProcessing();`

I was also satisfied with what Kayano explained, so I think I can write it that way.

Reply:
Thank you for your comment. I never thought there was such a way. This way of writing was cleaner, so I used it.🙇‍♀️